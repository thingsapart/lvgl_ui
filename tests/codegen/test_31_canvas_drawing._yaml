- type: style
  id: '@canvas_style'
  bg_color: '#ffffff'
  bg_opa: LV_OPA_COVER

- type: canvas
  add_style: ['@canvas_style', 0]
  lv_canvas_set_buffer: [NULL, 150, 100, LV_COLOR_FORMAT_NATIVE]
  center: []
  lv_canvas_fill_bg: [{ lv_palette_main: LV_PALETTE_GREY }, LV_OPA_50]
  lv_canvas_draw_rect: [5, 5, 50, 30, { radius: 5, bg_color: { lv_palette_main: LV_PALETTE_RED } }]
