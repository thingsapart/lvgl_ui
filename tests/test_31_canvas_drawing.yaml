# A canvas widget with a buffer and a simple rectangle drawn on it.
- type: style
  id: '@canvas_style'
  bg_color: '#ffffff'
  bg_opa: LV_OPA_COVER

- type: canvas
  add_style: ['@canvas_style', 0]
  buffer: [NULL, 150, 100, LV_COLOR_FORMAT_NATIVE]
  center: []
  fill_bg: [{ lv_palette_main: LV_PALETTE_GREY }, LV_OPA_50]
  draw_rect: [5, 5, 50, 30, { radius: 5, bg_color: { lv_palette_main: LV_PALETTE_RED } }]
