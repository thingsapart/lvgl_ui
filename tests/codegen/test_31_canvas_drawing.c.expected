/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // canvas_style: canvas_style_0 (style)
    lv_style_t* canvas_style_0 = malloc(sizeof(lv_style_t));

    lv_style_init(canvas_style_0);
    obj_registry_add("canvas_style", canvas_style_0);
    lv_style_set_bg_color(canvas_style_0, lv_color_hex(0xFFFFFF));
    lv_style_set_bg_opa(canvas_style_0, LV_OPA_COVER);

    // unnamed: canvas_1 (canvas)
    lv_obj_t* canvas_1 = lv_canvas_create(parent);

    lv_obj_add_style(canvas_1, canvas_style_0, 0);
    lv_canvas_set_buffer(canvas_1, "NULL", 150, 100, LV_COLOR_FORMAT_NATIVE);
    lv_obj_center(canvas_1);
    lv_canvas_fill_bg(canvas_1, lv_palette_main(LV_PALETTE_GREY), LV_OPA_50);
    // [GENERATOR HINT] Could not resolve property/method 'lv_canvas_draw_rect' for type 'canvas'.

}
